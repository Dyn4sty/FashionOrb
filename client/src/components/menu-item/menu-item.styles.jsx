import styled from 'styled-components'

export const MenuItemContainer = styled.div`
height: ${({ size }) => (size ? '380px' : '240px')};
min-width: 30%;
height: 350px;
flex: 1 1 auto;
display: flex;
align-items: flex-end;
justify-content: flex-end;
background-position: center;
background-size: cover; 
margin: 0 7.5px 15px;
overflow: hidden;

&:hover {
    cursor: pointer;
    & .background-image{
        transform: scale(1.1);
        transition: transform 6s cubic-bezier(0.25, 0.46, 0.45, 0.94);
    }
}
&:first-child {
margin-right: 7.5px;
}
&:last-child {
margin-left: 7.5px;
}
`;
MenuItemContainer.displayName = 'MenuItemContainer';

export const BackgroundImageContainer = styled.div`
    height: 100%;
    width: 100%;
    background-blend-mode: overlay;
    background-color: #ffffff36; 
    background-size: cover ;
    background-position: center;
background-image: url(${({ imageUrl }) => imageUrl});
`;
BackgroundImageContainer.displayName = "BackgroundImageContainer"

export const ContentContainer = styled.div`
    height: 90px;
    padding: 0 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    position: absolute ;
`;

export const ContentTitle = styled.h1`
    font-weight: 800;
    margin-bottom: 6px;
    font-size: 23px;
    text-transform: capitalize;
    color: #000;
`;

export const ContentSubtitle = styled.h2`
    font-weight: lighter;
    font-size: 13px;
    color: #000;
`;
