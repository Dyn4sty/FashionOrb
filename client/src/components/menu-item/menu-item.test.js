import React from "react";
import { shallow } from "enzyme";

import { MenuItem } from "./menu-item";

describe("MenuItem", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  const linkUrl = "shop/hats";
  const size = "large";
  const imageUrl = "testimage";

  beforeEach(() => {
    mockMatch = {
      url: "/",
    };
    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      title: "hats",
      imageUrl,
      size,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should render MenuItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push with the right string when MenuItemContainer clicked", () => {
    wrapper.find("MenuItemContainer").simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it("should pass size to MenuItemContainer as the prop size", () => {
    expect(wrapper.find("MenuItemContainer").prop("size")).toBe(size);
  });
  it("should pass imageUrl to BackgroundImageContainer as the prop ImageUrl", () => {
    expect(wrapper.find("BackgroundImageContainer").prop("imageUrl")).toBe(
      imageUrl
    );
  });
});
