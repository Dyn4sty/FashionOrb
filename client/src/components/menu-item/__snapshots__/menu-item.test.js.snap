// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render MenuItem component 1`] = `
<MenuItemContainer
  onClick={[Function]}
  size="large"
>
  <BackgroundImageContainer
    className="background-image"
    imageUrl="testimage"
  />
  <styled.div
    className="content"
  >
    <styled.h1>
      HATS
    </styled.h1>
    <styled.h2>
      SHOP NOW
    </styled.h2>
  </styled.div>
</MenuItemContainer>
`;
